src/api/app/controllers/staging/staged_requests_controller.rb

```diff
@@ -53,12 +53,28 @@ def destroy
 
   private
 
-  def request_numbers
-    xml_hash.elements('number')
+  def set_request_numbers
+    @request_numbers = @parsed_xml.elements('number')
+    return if @request_numbers.present?
+
+    render_error(
+      status: 400,
+      errorcode: 'invalid_request',
+      message: 'Error while parsing the numbers of the requests'
+    )
   end
 
-  def xml_hash
-    Xmlhash.parse(request.body.read) || {}
+  def set_xml_hash
+    request_body = request.body.read
+    @parsed_xml = Xmlhash.parse(request_body) if request_body.present?
+    return if @parsed_xml
+
+    error_options = if request_body.present?
+                      { status: 400, errorcode: 'invalid_xml_format', message: 'XML format is not valid' }
```

Comment:
What is not valid and what can I do about this?

Reply:
Being the mse
@hennevogel:
> What is not valid and what can I do about this?

Being the message "XML format is not valid", what is not valid is the XML format. What can one do about this: fix the XML format.

Do you think the message is not clear enough? What do you propose instead?

Reply:
Hm, I guess real validation with `Suse::Validator` but you are right, not now.